src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2755,10 +2758,10 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                     nativePdbWriter = new Cci.PdbWriter(pePdbFilePath, testSymWriterFactory, deterministic ? moduleBeingBuilt.PdbChecksumAlgorithm : default);
                 }
 
-                Func<Stream> getPortablePdbStream =
+                Func<Stream?>? getPortablePdbStream =
                     moduleBeingBuilt.DebugInformationFormat != DebugInformationFormat.PortablePdb || pdbStreamProvider == null
                     ? null
-                    : (Func<Stream>)(() => ConditionalGetOrCreateStream(pdbStreamProvider, metadataDiagnostics));
+                    : (Func<Stream?>?)(() => ConditionalGetOrCreateStream(pdbStreamProvider, metadataDiagnostics));
```

Comment:
>? [](start = 36, length = 1)

Minor: `?` should not be necessary.

Reply:
Which of the `?`

Reply:
The second `?` - the top-level nullability.

---
In reply to: [359951906](https://github.com/dotnet/roslyn/pull/40493#discussion_r359951906) [](ancestors = 359951906)

Reply:
The first clause of the conditional explicitly returns `null`

---
In reply to: [359952488](https://github.com/dotnet/roslyn/pull/40493#discussion_r359952488) [](ancestors = 359952488,359951906)

Reply:
Comment was for the type in the cast, not the local.

---
In reply to: [359988108](https://github.com/dotnet/roslyn/pull/40493#discussion_r359988108) [](ancestors = 359988108,359952488,359951906)